pyanaconda/payload/dnfpayload.py

```diff
@@ -566,13 +555,23 @@ def _apply_selections(self):
         # feed it to DNF
         try:
             # install_specs() returns a list of specs that appear to be missing
-            missing_specs = self._base.install_specs(install=include_list, exclude=exclude_list)
-            if missing_specs:
-                if self.data.packages.handleMissing == KS_MISSING_IGNORE:
-                    log.info("ignoring missing package/group/module specs due to --ingoremissing flag in kickstart: %s", missing_specs)
-                else:
-                    log.debug("install_specs() reports that some specs are missing: %s", missing_specs)
-                    raise InstallSpecsMissing(missing_specs)
+            self._base.install_specs(install=include_list, exclude=exclude_list)
+        except dnf.exceptions.MarkingErrors as e:
+            log.debug("install_specs(): some packages, groups or modules are missing or broken")
+            if e.no_match_pkg_specs:
+                log.debug("install_specs(): missing packages: %s", e.no_match_pkg_specs)
+            if e.no_match_group_specs:
+                log.debug("install_specs(): missing groups or modules: %s", e.no_match_group_specs)
+            if e.error_pkg_specs:
+                log.debug("install_specs(): broken packages: %s", e.error_pkg_specs)
+            if e.error_group_specs:
+                log.debug("install_specs(): broken groups or modules: %s", e.error_group_specs)
```

Comment:
Please, replace this code with `log.debug("install_specs(): %s", e)` and ask DNF to provide a nice string representation of this error.

Reply:
I think we should keep this until DNF provides the improved string representation of the exception (I'll ask them for it) as the the information about missing or broken groups/packages/modules is important for debugging and should always be logged.

Reply:
Ok.